integration/yarn/src/test/java/alluxio/yarn/ClientTest.java

```diff
@@ -0,0 +1,96 @@
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.yarn;
+
+import org.apache.hadoop.yarn.api.protocolrecords.GetNewApplicationResponse;
+import org.apache.hadoop.yarn.api.records.Resource;
+import org.apache.hadoop.yarn.client.api.YarnClient;
+import org.apache.hadoop.yarn.client.api.YarnClientApplication;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import static org.mockito.Mockito.*;
+
+import alluxio.Configuration;
+import alluxio.Constants;
+import alluxio.PropertyKey;
+import alluxio.exception.ExceptionMessage;
+
+
+/**
+ * Unit tests for {@link Client}.
+ * TODO: ALLUXIO-1503: add more unit test for alluxio.yarn.Client
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({Client.class, YarnClient.class})
+public final class ClientTest {
+
+  private YarnClient mYarnClient;
+
+  @Rule
+  public ExpectedException mThrown = ExpectedException.none();
+
+  @Before
+  public void before() {
+    // Mock Yarn client
+    PowerMockito.mockStatic(YarnClient.class);
+    mYarnClient = (YarnClient) Mockito.mock(YarnClient.class);
+    Mockito.when(YarnClient.createYarnClient()).thenReturn(mYarnClient);
+  }
+
+  /**
+   * Tests that the main method properly reads args.
+   */
+  @Test(timeout = 10000)
+  public void mainTest() throws Exception {
+    Client mockClient = spy(new Client());
+    Mockito.doNothing().when(mockClient).run();
+    PowerMockito.whenNew(Client.class)
+        .withNoArguments()
+        .thenReturn(mockClient);
+
+    Client.main(new String[] {
+        "-resource_path", "hdfs://test",
+        "-am_memory", "1024",
+        "-am_vcores", "2"});
+    PowerMockito.verifyNew(Client.class).withNoArguments();
+
+    Mockito.verify(mockClient).run();
+  }
+
+  @Test(timeout = 10000)
+  public void notEnoughResources() throws Exception {
+    int masterMemInMB = (int) (Configuration.getBytes(
+        PropertyKey.INTEGRATION_MASTER_RESOURCE_MEM) / Constants.MB);
+    YarnClientApplication mockYarnClientApplication = mock(YarnClientApplication.class);
+    doReturn(mockYarnClientApplication).when(mYarnClient).createApplication();
+    GetNewApplicationResponse mockNewApplicationResponse = mock(GetNewApplicationResponse.class);
+    doReturn(mockNewApplicationResponse).when(mockYarnClientApplication)
+        .getNewApplicationResponse();
+    int maxAllocationMem = masterMemInMB/2;
+    Resource resource = Resource.newInstance(maxAllocationMem, 1);
+    doReturn(resource).when(mockNewApplicationResponse).getMaximumResourceCapability();
```

Comment:
We should test all resource checks, giving us 6 tests (memory/cores for AM, master, and workers). We can move the part of this method before this line to a helper method shared by all 6 tests. The method would just take the `Resource` which Yarn should pretend to have as its maximum resource capability.